src/base/emu-i386/simx86/interp.c

```diff
@@ -3343,6 +3341,19 @@ intop3b:		{ int op = ArOpsFR[D_MO(opc)];
 			P0 = PC;
 			CODE_FLUSH();
 		}
+		if (CEmuStat & CeS_MOVSS) {
+			/* following non-compiled (sim or protected mode)
+			   mov ss / pop ss only */
+			if (!(CEmuStat & CeS_INHI)) {
+				// directly following mov ss / pop ss
+				CEmuStat |= CeS_INHI;
+				CEmuStat &= ~CeS_TRAP;
```

Comment:
Do you want to reset CeS_MOVSS here as well?
Or it just doesn't matter?

Reply:
It does matter. The way I programmed it is that for non-compiled code (protected mode or sim), CeS_INHI is only ever set if CeS_MOVSS is also set, so we can't just have CeS_INHI and no CeS_MOVSS (this can be later extended to CeS_MOVSS|CeS_STI)).

This way also both the if and else branch here trigger in the special case. If I reset CeS_MOVSS, then the CeS_INHI needs to be reset in a general else branch, ie. on *every* interpreter iteration instead of just when (CeS_MOVSS|CeS_INHI) are both set.